pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -457,13 +460,47 @@ var (
 	errNewServiceAccount = errors.New("service account change")
 )
 
+// WorkloadUpdate is called when workload labels/annotations are updated.
+func (s *DiscoveryServer) WorkloadUpdate(id string, labels map[string]string, annotations map[string]string) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	w, f := s.WorkloadsById[id]
+	if !f {
+		// First time this workload has been seen. Likely never connected, no need to
+		// push
+		s.WorkloadsById[id] = &Workload{
+			Labels:      labels,
+			Annotations: annotations,
+		}
+		return
+	}
+	if reflect.DeepEqual(w.Labels, labels) {
```

Comment:
This is incorrect. Map iteration order is non-deterministic right? Order of labels could be different but still same labels. I think the model.Labels has some utility functions to compare two sets of labels

Reply:
My undestanding was that DeepEqual is aware of maps - looked at the code and seems to check if both have same keys and same values, regardless of order.

https://stackoverflow.com/questions/18208394/testing-equivalence-of-maps-golang

DeepEqual doc:
"Map values are deeply equal when all of the following are true: they are both nil or both non-nil, they have the same length, and either they are the same map object or their corresponding keys (matched using Go equality) map to deeply equal values."